### ISSUE TITLE
salt-cloud - enable mem-hotadd to my salt cloud profile. Extra config array ignored

### ISSUE THREAD

**Issue by roddergreg** [saltstack/salt#56144]
salt-cloud for vmware not function properly. When specifying `mem.hotadd` to the extra-config array value is ignored. Does not work in Template or Map file

Salt-Cloud Profile:
```
golden:
  provider: vcenter60
  clonefrom: 'template-rhel7-generic-salt-cloud'
  ## Optional arguments
  num_cpus: 4
  memory: 16GB
  devices:
   network:
      Network adapter 1:
       name: 'VLAN XX Internal Tools Host'
       switch_type: distributed
       adapter_type: vmxnet3
       ip: 10.104.XX.XXX
       gateway: [10.XXX.XX.X]
       subnet_mask: 255.255.255.0
       domain: mydomain.com

  domain: mydomain.com

  dns_search:
    -  mydomain.com

  dns_servers:
    - 10.104.8.201
    - 10.104.8.202

  resourcepool: 'Tst Resource Pool 
  cluster: 'High Clock'
  datastore: 'General Purpose Cluster'
  datacenter: DataCenter1
  template: False
  power_on: True
  mem.hotadd: 'yes'
  extra_config:
    disk.EnableUUID: 'yes'
    **mem.hotadd: 'yes'**
  annotation: Created by Salt-Cloud

  deploy: True
  customization: True
  private_key: /root/.ssh/id_rsa
  ssh_username: root
  minion:
    master: salt.mydomain.com
    append_domain: mydomain.com
    backup_mode: minion
    recon_randomize: True
    log_level_logfile: info
```

Salt-cloud Template:
```
golden:
 - gold0:
     devices:
      network:
        Network adapter 1:
          name: 'VLAN XX Internal Tools Host'
          switch_type: distributed
          adapter_type: vmxnet3
          ip: 10.104.XX.XXX
          gateway: [10.104.XXX.1]
          subnet_mask: 255.255.255.0
```

Run `salt-cloud -m gold0`  vm is successfully created but the configuration option for `mem.hotadd` is ignored.  I have to power the server off turn on `mem.hotadd` manually and power back on to get this enabled. 

From information on the web and some people from the Salt-cloud slack group.  Vmware made a change that needs to be captured in the vmware salt-cloud provider. 

Supporting Documentation:
https://communities.vmware.com/thread/619066

### Versions Report
```
Salt Version:
            Salt: 2019.2.3
 
Dependency Versions:
 Apache Libcloud: 2.2.1
            cffi: 1.6.0
        cherrypy: 5.6.0
        dateutil: 1.5
       docker-py: Not Installed
           gitdb: Not Installed
       gitpython: Not Installed
           ioflo: Not Installed
          Jinja2: 2.7.2
         libgit2: 0.26.3
         libnacl: Not Installed
        M2Crypto: 0.31.0
            Mako: 0.8.1
    msgpack-pure: Not Installed
  msgpack-python: 0.5.6
    mysql-python: Not Installed
       pycparser: 2.14
        pycrypto: 2.6.1
    pycryptodome: Not Installed
          pygit2: 0.26.4
          Python: 2.7.5 (default, Jun 11 2019, 14:33:56)
    python-gnupg: Not Installed
          PyYAML: 3.11
           PyZMQ: 15.3.0
            RAET: Not Installed
           smmap: Not Installed
         timelib: Not Installed
         Tornado: 4.2.1
             ZMQ: 4.1.4

System Versions:
            dist: redhat 7.7 Maipo
          locale: UTF-8
         machine: x86_64
         release: 3.10.0-1062.9.1.el7.x86_64
          system: Linux
         version: Red Hat Enterprise Linux Server 7.7 Maipo
```

Thank you everyone for your assistance!!

Best,

Greg

**Comment by nicholasmhughes** [comment]
@roddergreg , grab this updated vmware cloud provider and see if it works for you.

https://github.com/nicholasmhughes/salt/blob/fix_vmware_hotadd_enabled/salt/cloud/clouds/vmware.py

Drop it in `/srv/salt/_clouds/` and run `salt-run saltutil.sync_clouds` to test.

**Comment by roddergreg** [comment]
updated provider as instructed.  Looks like it did not work. 

![image](https://user-images.githubusercontent.com/3171527/74743836-1e173780-522f-11ea-91ba-69e696ccb3c6.png)

my normal vmware.py lives in  /usr/lib/python2.7/site-packages/salt/utils/vmware.py

Did this synchronize with the salt-run?
Thanks,
Greg

**Comment by roddergreg** [comment]
I had pulled down the vmware.py without the updated pr information. I pulled the new file and ran the sync and it works perfectly for mem_hot_add.  Here is my screen from a running vm.  The Memory is no longer gray and the Memory hot plug is checked!

![image](https://user-images.githubusercontent.com/3171527/74778703-1de85d80-526a-11ea-8c0d-23685856d5ff.png)

This PR works as expected. Thank you Nicholas!!
